src/util/ttl-cache.ts

```diff
@@ -0,0 +1,17 @@
+type Getter<T> = () => Promise<T>
+type GetterFunction<T> = (getter: Getter<T>) => Promise<T>
+
+export function makeTtlCache<T>(ttl: number): GetterFunction<T> {
```

Comment:
Why is this curried in this way? Shouldn't the getter and TTL values both be provided at construction time, since the cache can't do anything useful without both?

Reply:
You're right. I forgot to change this. Originally this utility was curried this way to allow for closures over state local to a particular call-site. This was because the IO was passed as params in `edge-core-js`. In this repo, the IO is available outside of the call-site, so I technically don't need to curry anymore. Although, the API could still support this; another way of looking at this is that a "ttl cache" is just a function which returns the previous value cached if the TTL isn't reached

Reply:
I've updated the PR with a fixup commit to hopefully make the API clearer in its intention and use. To reiterate, a `TtlCache` is a "cached getter function"; you pass in a getter and it returns either the already cached value (if the ttl hasn't expired) or it calls your getter function to update the value. The getter function must return the same type as it is defined at the `makeTtlCache` declaration.